ddtrace/tracer/spancontext.go

```diff
@@ -414,10 +415,86 @@ func (t *trace) finishedOne(s *span) {
 	if hn := tr.hostname(); hn != "" {
 		s.setMeta(keyTracerHostname, hn)
 	}
+	setPeerService(s, tr.config)
```

Comment:
I just tested this for a service in `dd-go`, and this never get executed. In almost all cases the `peer.service` needs to be set on a leaf span, but because of the condition at line 404 `if len(t.spans) != t.finished { return }`, this function is never reached.